core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -404,8 +404,11 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
    * does not fail, but succeeds with the value returned by the left or right
    * function passed to `fold`.
    */
-  final def fold[B](err: E => B, succ: A => B): ZIO[R, Nothing, B] =
-    foldM(new ZIO.MapFn(err), new ZIO.MapFn(succ))
+  final def fold[B](
+    @deprecatedName('err) failure: E => B,
+    @deprecatedName('succ) success: A => B
+  ): ZIO[R, Nothing, B] =
```

Comment:
I think we can eliminate the `deprecatedName`. is that ok for you?

Reply:
For me yes. Thought that may somebody is using the names and it was easy to add. (Not totally trivial, since `deprecatedName` is different in every Scala-Version).
I removed the annotations and ammended the commit.